src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/IProjectTreeExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.VisualStudio.Shell;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal static class IProjectTreeExtensions
+    {
+        public static HierarchyId GetHierarchyId(this IProjectTree tree) =>
+            new HierarchyId(tree.IsRoot() ? VSConstants.VSITEMID_ROOT : unchecked((uint)tree.Identity.ToInt32()));
```

Comment:
There's implicit conversion from uint -> HierarchyId and vice versa. Secondly, I'd just simply the cast a little to something like:

```
return tree.IsRoot() ? HierarchyId.Root : checked((uint)tree.Identity))
```

We probably want to know if tree.Identity represents an negative number.